### PR TITLE
[Pages Publish] Reduced parallelism and added KV write retries

### PR DESCRIPTION
**Pull request by geelen**

### CODE REVIEW

File: packages/wrangler/src/pages.tsx [hunk 1]
```diff
@@ -1198,15 +1198,33 @@ const createDeployment: CommandModule<
         },
         base64: true,
       }));
+
+      let attempts = 0;
+      const doUpload = async (): Promise<void> => {
+        try {
+          return await fetchResult(`/pages/assets/upload`, {
```
Comment (sidharthachatterjee) [inline packages/wrangler/src/pages.tsx:1205]: If this returns with `success: false`, does that throw?

**Review by sidharthachatterjee — APPROVED** [review]
Thank you for doing this!

File: packages/wrangler/src/pages.tsx (hunk 1 above)
Comment (geelen) [reply packages/wrangler/src/pages.tsx:1205]: No. Weirdly. I tried `new Response` with status 500, all kinds of things. The only way I could get this to actually loop was an exception within the test :/

Comment (sidharthachatterjee) [reply packages/wrangler/src/pages.tsx:1205]: Oh well

Comment (petebacondarwin) [reply packages/wrangler/src/pages.tsx:1205]: `setMockResponse()` always returns a response with `success: true`.
If you want to simulate returning something else then you should consider `setMockRawResponse()`.

Comment (petebacondarwin) [reply packages/wrangler/src/pages.tsx:1205]: But rest assured, `fetchResponse()` will thrown if `success: false`.

Comment (geelen) [reply packages/wrangler/src/pages.tsx:1205]: Aha now `setMockRawResponse` makes sense to me! Fixed now.

**Pull request merged**